subprojects/internal-instrumentation-processor/src/main/java/org/gradle/internal/instrumentation/processor/codegen/jvmbytecode/InterceptJvmCallsGenerator.java

```diff
@@ -250,9 +250,16 @@ private static void documentInterceptorGeneratedCode(CallInterceptionRequest req
     }
 
     private static CodeBlock matchOpcodeExpression(CallableInfo interceptedCallable) {
-        CodeBlock result = interceptedCallable.getKind() == CallableKindInfo.STATIC_METHOD ? CodeBlock.of("opcode == $T.INVOKESTATIC", Opcodes.class) :
-            interceptedCallable.getKind() == CallableKindInfo.INSTANCE_METHOD ? CodeBlock.of("opcode == $T.INVOKEVIRTUAL", Opcodes.class) :
-                interceptedCallable.getKind() == CallableKindInfo.AFTER_CONSTRUCTOR ? CodeBlock.of("opcode == $T.INVOKESPECIAL", Opcodes.class) : null;
+        CodeBlock result = null;
+        if (interceptedCallable.getKind() == CallableKindInfo.STATIC_METHOD) {
+            result = CodeBlock.of("opcode == $T.INVOKESTATIC", Opcodes.class);
+        } else if (interceptedCallable.getKind() == CallableKindInfo.INSTANCE_METHOD && interceptedCallable.getOwner().isInterceptSubtypes()) {
```

Comment:
Should we match the `invokeinterface` opcode even if `isInterceptSubtypes` is false? In my other branch, I found that we didn't and added this unconditionally.

Reply:
[`a7415d5` (#24949)](https://github.com/gradle/gradle/pull/24949/commits/a7415d56acf2871c05399188f33a9f1124ff351f)

Reply:
Applied that to my branch.